Src/IronPython/Runtime/Operations/FloatOps.cs

```diff
@@ -53,39 +53,66 @@ public static partial class DoubleOps {
                 return res;
 
             throw PythonOps.ValueError($"could not convert string to {type.Name}: {PythonOps.Repr(context, x)}");
+        }
+
+        [StaticExtensionMethod]
+        public static object __new__(CodeContext/*!*/ context, PythonType cls, object x) {
+            object value = NewFloat(context, TypeCache.Double, x);
+
+            if (cls == TypeCache.Double) {
+                return value;
+            } else {
+                return cls.CreateInstance(context, value);
+            }
+        }
+
+        internal static bool TryToFloat(CodeContext context, object/*?*/ value, out double result) {
+            if (value is double d) {
+                result = d;
+            } else if (value is int i) {
+                result = i;
+            } else if (value is BigInteger bi) {
+                result = (double)bi;
```

Comment:
I guess these cast is where testing fails. The BigInteger to double cast returns infinity instead of overflowing. There's a `MathUtils.ToFloat64` that throws (but we should consider a helper at the ipy level so we can match the error message).

Reply:
Indeed. The fast track just cut too many corners. `__float__` and `__complex__` on `int` do it correctly, whith a call to `MathUtils.ToFloat64`. Funny thing, I started working on `complex.__new__` as a spin-off from an investigation whether `int.__complex__` can be safely removed (it can, assuming that `complex.__new__` works correctly) and then completely forgot about that.

I like the idea of our own helper class with a Python-specific error message.